### PR TITLE
Added render.setCullMode

### PR DESCRIPTION
**Pull request by CornerPin**
Added **render.setCullMode**. Cull mode is reset to _MATERIAL_CULLMODE_CCW_ in cleanup hook, so render can not be broken.

### CODE REVIEW

**Comment by Damianu** [comment]
As I said on discord:
```
Sizes are a bad idea
RT cant be removed
And it gives you option to create various sizes of rts to spam texture memory
Thats why starfall reuses it
```
+ Why do you need different sizes of RT? Just use a part of it.

File: lua/starfall/libs_cl/render.lua [hunk 1]
```diff
@@ -914,6 +930,19 @@ function render_library.setFilterMin(val)
 	render.PushFilterMin(val)
 end
 
+--- Changes the cull mode
+-- @param mode Cull mode. 0 for counter clock wise, 1 for clock wise
+function render_library.setCullMode(mode)
+	if not SF.instance.data.render.isRendering then SF.Throw("Not in a rendering hook.", 2) end
+	SF.CheckLuaType(mode, TYPE_NUMBER)
+
+	if mode >= 0 and mode <= 1 then
```
Comment (Damianu) [inline lua/starfall/libs_cl/render.lua]: I think its way cleaner to do
```
mode = mode == 1 and 1 or 0
render.CullMode(mode)
```
and remove typecheck

**Comment by CornerPin** [comment]
It won't use much more memory than at the moment
memory
It makes things easier, you don't need to make scaling matrices or other stuff.
Should I remove this?

**Comment by Damianu** [comment]
The problem isnt memory increase on normal usage. It's about avoiding limit by using different sizes and recreating them(because you can make RTLimit*possibleSizes, right now its just RT Limit).

**Comment by Damianu** [comment]
I like CullMode idea tho.

**Pull request merged**